flock_agent/daemon/daemon.py

```diff
@@ -357,3 +361,96 @@ def log(self, request, response, time):
         # Change permissions of unix socket
         os.chmod(self.unix_socket_path, 0o660)
         os.chown(self.unix_socket_path, 0, self.gid)
+
+    async def autoupdate_loop(self):
+        # Autoupdate is only available for macOS right now; Linux uses package managers
+        if Platform.current() != Platform.MACOS:
+            return
+
+        update_check_delay = 43200  # 12 hours
+
+        while True:
+            self.c.log("Daemon", "autoupdate_loop", "Checking for updates")
+
+            try:
+                # Query github for the latest version of Flock Agent
+                r = requests.get(
+                    "https://api.github.com/repos/firstlookmedia/flock-agent/releases/latest"
+                )
+                release = r.json()
+                latest_version = release["tag_name"].lstrip("v")
+                self.c.log(
+                    "Daemon",
+                    "autoupdate_loop",
+                    f"installed version: {self.c.version}, latest version: {latest_version}",
+                )
+                if latest_version <= self.c.version:
```

Comment:
We should compare with a version comparison library, no? Otherwise 0.9 will be greater than 0.10.

Reply:
Good call. I should be comparing this with a version comparison library.